var/spack/repos/builtin/packages/openfoam-com/package.py

```diff
@@ -292,6 +292,10 @@ class OpenfoamCom(Package):
 
     provides('openfoam')
     depends_on('mpi')
+
+    # Newest version requires openmpi+thread_multiple for collated output
+    conflicts('openmpi~thread_multiple', when="^openmpi")
```

Comment:
> The newest openfoam version

should appear in the conflict somehow, i.e. `when="@x.y.z:"`

Reply:
True, but then I'd need similar logic in the run-time hack. I'd just like to get it sorted in any way possible first and figured that this way people would have a concrete example to play with on their own desktop to see how the spec resolution is behaving/misbehaving.